architecture-decision-records/0008-keycloak-user-administrators-configuration.md

```diff
@@ -0,0 +1,104 @@
+* *User data protection*: ensure user data is secure
+
+## Decision
+
+Decided to define user administrators in the "master" realm. 
+
+There is some benefit from having the user administrators in the master realm as such users will not have access to themselves and cannot give themselves elevated privileges.
+
+Having user administrators in the master realm also provides a clear demarcation between transferring body users which will be defined in the TDR realm users. This provides additional protection against inadvertently giving incorrect permissions to transferring body users.
+
+## TDR User Administrators Configuration
+
+User administrators will be users that have permission to manage other users. 
+
+This will include transferring body users and internal TNA users.
+
+"Groups" will be used to define the permissions. This will allow easier administration of the permissions, as the permission will be defined by the group and keycloak users can be added or removed to the groups.
+
+### Options considered
+
+#### Option 1: Define user administrators in master and TDR realms
+
+The transferring body user administrators will be defined in the TDR realm, with the "super" user administrators defined in the master realm
+
+* **Master Realm**
+  * *super_user_administrator* group:   
+    * Access to all users/groups within the master realm only    
+    * Has access to self 
+* **TDR Realm**
+  * *user_administrator* group:  
+    * Access to all users/groups within the tdr-realm only    
+    * Access to self
+  * *transferring_body_user* group: 
+    * Access to TDR application only
+    * Access to series belonging to transferring body
+
+With this configuration the transferring body user administrators and super user administrators can give themselves limited elevated privileges.
+
+#### Option 2: Define TDR user administrators in master realm
+
+Both transferring body user administrators and "super" user administrators will be defined in the master realm
+
+* **Master Realm**
+  * *super_user_administrator* group:   
+    * Access to all users/groups within the master realm only (this includes the *tdr_user_administrator* as this is held within the master realm)    
+    * Has access to self    
+  * *tdr_user_administrator* group:  
+    * Access to all users/groups within the TDR realm only    
+    * No access to self  
+* **TDR Realm**
+  * *transferring_body_user* group: 
+    * Access to TDR application
+    * Access to series belonging to transferring body
+
+With this configuration the super user administrators can give themselves limited elevated privileges.
+
+## Keycloak User Configuration Background
+
+### Realms
+
+Keycloak user the concept of "realms".
+
+A realm manages a set of users, credentials, roles, and groups. A user belongs to and logs into a realm. 
+
+Realms are isolated from one another and can only manage and authenticate the users that they control.
+
+#### "Master" Realm
```

Comment:
Do you know if it's possible to rename the master realm, e.g. to Main or Admin? The use of "Master" in this context has racist origins and connotations, and there's currently a [big push to switch to other terms](https://www.bbc.co.uk/news/technology-53050955). This seems like a good time to do it, given that we're reconfiguring Keycloak anyway. 🙂

Reply:
This does appear to be the case, in the standalone-ha.xml:

`<master-realm-name>master</master-realm-name>`

I'm going to be doing some changes in the tdr-auth-server so can probably implement this as well. Will call it: admin